### PR TITLE
BngSimulator: respect cleanup flag in `__init__`

### PR DESCRIPTION
**Pull request by alubbock**
The cleanup flag in BngSimulator's `__init__()` was being ignored;
only the value specified to `run()` was used. The `__init__()`
cleanup flag is now treated as a default, which can be overridden
on a per simulation basis in `run()` if required.

### CODE REVIEW

File: pysb/simulator/bng.py [hunk 1]
```diff
@@ -152,7 +153,8 @@ def run(self, tspan=None, initials=None, param_values=None, n_runs=1,
                               verbose=verbose_bool,
                               output_dir=output_dir,
                               output_prefix=output_file_basename,
-                              cleanup=cleanup,
+                              cleanup=self.cleanup if cleanup is None
```
Comment (jmuhlich) [inline pysb/simulator/bng.py]: The formatting here is a little unfortunate. Maybe just normalize the value of `cleanup` with `if cleanup is None: cleanup = self.cleanup` up with all the other parameter checking?

Comment (alubbock) [reply pysb/simulator/bng.py]: Sounds good. Done.

File: pysb/simulator/bng.py [hunk 2]
```diff
@@ -95,6 +95,9 @@ def run(self, tspan=None, initials=None, param_values=None, n_runs=1,
                                       _run_kwargs=locals()
                                       )
 
+        if cleanup is None:
+            clean = self.cleanup
```
Comment (jmuhlich) [inline pysb/simulator/bng.py]: I think `clean` is a typo, should be `cleanup`.

Comment (alubbock) [reply pysb/simulator/bng.py]: Gah, indeed! Fixed.

**Pull request merged**